docs/Customization/Dsptools-Blocks.rst

```diff
@@ -0,0 +1,118 @@
+.. _dsptools-blocks:
+
+Dsptools is a Chisel library that aids in writing custom signal processing accelerators. It does this by:
+* Giving types and helpers that allow you to express mathematical operations more directly.
+* Typeclasses that let you write polymorphic generators, for example an FIR filter generator that works for both real- and complex-valued filters.
+* Structures for packaging DSP blocks and integrating them into a rocketchip-based SoC.
+* Test harnesses for testing DSP circuits, as well as VIP-style drivers and monitors for DSP blocks.
+
+The `Dsptools <https://github.com/ucb-bar/dsptools/>`_ repository has more documentation.
+
+
+Dsptools Blocks
+===============
+A ``DspBlock`` is the basic unit of signal processing functionality that can be integrated into an SoC.
+It has a AXI4-stream interface and an optional memory interface.
+The idea idea is that these ``DspBlocks`` can be easily designed, unit tested, and assembled lego-style to build complex functionality.
+A ``DspChain`` is one example of how to assemble ``DspBlocks``, in which case the streaming interfaces are connected serially into a pipeline, and a bus is instatiated and connected to every block with a memory interface.
+ 
+This project has example designs that integrate a ``DspBlock`` to a rocketchip-based SoC as an MMIO peripheral. The custom ``DspBlock`` has a ``ReadQueue`` before it and a ``WriteQueue`` after it, which allow memory mapped access to the streaming interfaces so the rocket core can interact with the ``DspBlock``.  This section will primarily focus on designing Tilelink-based peripherals. However, through the resources provided in Dsptools, one could also define an AXI4-based peripheral by following similar steps. Furthermore, the examples here are simple, but can be extended to implement more complex accelerators, for example an `OFDM baseband <https://github.com/grebe/ofdm>`_ or a `spectrometer <https://github.com/ucb-art/craft2-chip>`_.
```

Comment:
Is this an example of a DSPChain made of a single DSPBlock?

Reply:
I think a block diagram would be extremely helpful here (where do the ReadQueue and WriteQueue fit within the DSPBlock/DSPChain abstractions, and where does Rocket and the memory system fit

Reply:
This is not actually written with a DspChain. I could write it to be done that way, but I think it is easiest to start with just one DspBlock and connect it as a peripheral.

Agree re: block diagram.

Reply:
I think if you explaining a concept, you want to have a minimal example of each part of the concept (i.e., a DSPBlock, and a DSPChain made of 2 DSPBlocks)

Reply:
I switched it over to using a dspchain, which I think makes the example more complete. I still need to go through the docs more to make sure they're still consistent with the code and add a block diagram